scalafix-cli/src/main/scala/scalafix/cli/Cli.scala

```diff
@@ -5,19 +5,43 @@ import scala.meta.Database
 import scala.meta.io.AbsolutePath
 import scalafix.cli.CliCommand.PrintAndExit
 import scalafix.cli.CliCommand.RunScalafix
+import scalafix.rewrite.ScalafixRewrites
+import caseapp.Name
+import caseapp.core.Arg
+import caseapp.core.Messages
 import caseapp.core.WithHelp
 import com.martiansoftware.nailgun.NGContext
 import metaconfig.Configured.NotOk
 import metaconfig.Configured.Ok
 
 object Cli {
   import ArgParserImplicits._
-  private val withHelp = OptionsMessages.withHelp
+  private val withHelp: Messages[WithHelp[ScalafixOptions]] =
+    OptionsMessages.copy(optionsDesc = "[options] [<file>...]").withHelp
   val helpMessage: String = withHelp.helpMessage +
-    s"""|
-        |Examples:
-        |  $$ scalafix --rewrites ProcedureSyntax Code.scala # write fixed file in-place
-        |  $$ scalafix --stdout --rewrites ProcedureSyntax Code.scala # print fixed file to stdout
+    s"""|Available rewrites: ${ScalafixRewrites.allNames.mkString(", ")}
+        |
+        |NOTE. The command line tool is mostly intended to be invoked programmatically
+        |from build-tool integrations such as sbt-scalafix. The necessary fixture to run
+        |semantic rewrites is tricky to setup manually.
+        |
+        |Scalafix chooses which files to fix according to the following rules:
+        |- when --syntactic is passed, then Scalafix looks for .scala files in the provided files/directories.
+        |- by default, looks for .semanticdb files and matches them back to the original files.
+        |  - if --classpath and --sourcepath are provided, then those are used to find .semanticdb files.
+        |  - otherwise, Scalafix will automatically look for META-INF/semanticdb directories from the
+        |    current working directory.
+        |
+        |Examples (semantic):
+        |  $$ scalafix # automatically finds .semanticdb files and runs rewrite configured in .scalafix.conf.
+        |  $$ scalafix <directory> # same as above except only run on files in <directory>
+        |  $$ scalafix --rewrites RemoveUnusedImports # same as above but run RemoveUnusedImports.
+        |  $$ scalafix --classpath <foo.jar:target/classes> --sourceroot <directory> # customize semanticdb.
+        |
+        |Examples (syntactic):
```

Comment:
That is an interesting idea. It's simple for provided rewrites but more tricky for classloaded rewrites. I think it's best to leave that for another PR, https://github.com/scalacenter/scalafix/issues/221